### PR TITLE
Hide to tray

### PR DESCRIPTION
**Pull request by simonwiles**
This is an implementation of the "hide-to-tray" functionality.  It is a more complete alternative to #279.

### CODE REVIEW

**Comment by oroulet** [comment]
Great to see some activity here!

File: qtodotxt/ui/controllers/main_controller.py [hunk 1]
```diff
@@ -404,8 +411,8 @@ def updateFilters(self):
         self._onFilterSelectionChanged(self._filters_tree_controller.view.getSelectedFilters())
 
     def toggleVisible(self):
-        if self.view.isMinimized():
+        if self.view.isMinimized() or self.view.isHidden():
             self.view.showNormal()
```
Comment (PKEv) [inline qtodotxt/ui/controllers/main_controller.py]: Not quite correct behavior in the sense that if the window is maximum, after minimize and restore from tray, the window will have normal size

**Comment by PKEv** [comment]
It's much better than what I did, thank you very much for the work

File: qtodotxt/ui/controllers/main_controller.py (hunk 1 above)
Comment (PKEv) [inline qtodotxt/ui/controllers/main_controller.py]: Offer `self.view.showNormal()` change to `self.view.show()`

File: qtodotxt/app.py [hunk 2]
```diff
@@ -27,7 +27,32 @@ def _initIcon(self):
         self.activated.connect(self._onActivated)
         self.setToolTip('QTodoTxt')
 
-    def _onActivated(self):
+        menu = QtWidgets.QMenu()
+        create_task_action = menu.addAction("Create New Task")
```
Comment (PKEv) [inline qtodotxt/app.py]: Offered with a picture:
`create_task_action = menu.addAction(getIcon('TaskCreate.png'), "Create New Task")`

File: qtodotxt/app.py [hunk 3]
```diff
@@ -27,7 +27,32 @@ def _initIcon(self):
...
+        create_task_action.triggered.connect(self._createTask)
+        toggle_visible_action = menu.addAction("Show/Hide Window")
+        toggle_visible_action.triggered.connect(self._controller.toggleVisible)
+        exit_action = menu.addAction("Exit")
```
Comment (PKEv) [inline qtodotxt/app.py]: Offered with a picture:
exit_action = menu.addAction(getIcon('ApplicationExit.png'), "Exit")

**Pull request merged**